sqlx-macros/src/derives.rs

```diff
@@ -0,0 +1,110 @@
+use proc_macro2::Span;
+use quote::quote;
+use syn::punctuated::Punctuated;
+use syn::token::{Gt, Lt, Where};
+use syn::{
+    parse_quote, Data, DataStruct, DeriveInput, Fields, PredicateType, Token, WhereClause,
+    WherePredicate,
+};
+
+pub(crate) fn expand_derive_encode(input: DeriveInput) -> syn::Result<proc_macro2::TokenStream> {
+    if let Data::Struct(DataStruct {
+        fields: Fields::Unnamed(fields),
+        ..
+    }) = &input.data
+    {
+        let fields = &fields.unnamed;
+        if fields.len() == 1 {
```

Comment:
You should be able to lift this to a pattern guard so you don't have to repeat the error at the bottom. Moves in patterns with guards have been supported since 1.39: https://github.com/rust-lang/rust/blob/master/RELEASES.md#language-1